### PR TITLE
DebugLink click tests

### PR DESCRIPTION
**Pull request by matejcik**
this implements click tests by itself

### CODE REVIEW

File: core/src/trezor/messages/MessageType.py [hunk 1]
```diff
@@ -67,6 +67,7 @@
 DebugLinkMemory = 111
 DebugLinkMemoryWrite = 112
 DebugLinkFlashErase = 113
+DebugLinkLayout = 9001
```
Comment (tsusanka) [inline core/src/trezor/messages/MessageType.py]: Does the 99 prefix make sense if the other Debug messages do not have it?

File: tests/background.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
+from concurrent.futures import ThreadPoolExecutor
```
Comment (tsusanka) [inline tests/background.py]: I don't like the filename, maybe `background_handler` ?

File: tests/conftest.py [hunk 3]
```diff
@@ -146,3 +148,23 @@ def pytest_runtest_setup(item):
...
         pytest.skip("Skipping altcoin test")
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # Make test results available in fixtures.
+    # See https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+    outcome = yield
+    rep = outcome.get_result()
+    setattr(item, f"rep_{rep.when}", rep)
```
Comment (tsusanka) [inline tests/conftest.py]: I'm unsure what is this doing and how it relates to the the click tests.

File: core/src/trezor/messages/MessageType.py (hunk 1 above)
Comment (tsusanka) [reply core/src/trezor/messages/MessageType.py]: Hm this is quite a mess. I'd vote to change the other DebugLink message then, but maybe it does more harm than good. Any opinions on this @prusnak?

File: tests/conftest.py (hunk 3 above)
Comment (matejcik) [reply tests/conftest.py]: it makes the result of the test available in a fixture ;)

The `device_handler` fixture raises an error if you forget to check status of your background task. However, if your test fails, it probably never gets to the line that does the checking. In that case you'd get a failure *and* an error. That seemed less than friendly, so instead the error is only raised if the test succeeds.

File: core/src/trezor/messages/MessageType.py (hunk 1 above)
Comment (matejcik) [reply core/src/trezor/messages/MessageType.py]: the problem with changing existing debuglink codes is that it will break upgrade-tests

Comment (tsusanka) [reply core/src/trezor/messages/MessageType.py]: Ay. Hmm, let's keep it, I think it's more of an OCD thing anyway.

File: tests/conftest.py (hunk 3 above)
Comment (tsusanka) [reply tests/conftest.py]: I didn't realize what is a fixture in this context. Thanks.

Comment (matejcik) [reply tests/conftest.py]: added a comment that says a little about this :)

**Comment by matejcik** [comment]
the Shamir persistence test is currently failing, because `read_layout()` is not returning anything in the CI environment. I suspect that it is a race condition between starting of default layout and handling of the debuglink message. Not sure how to solve it ATM.

**Comment by tsusanka** [comment]
> the Shamir persistence test is currently failing, because read_layout() is not returning anything in the CI environment.

Does it work on your computer? Because surprisingly it seems to fail on CI but stalls on mine.

**Comment by matejcik** [comment]
yes, it works fine for me. I don't see a reason for stalling - maybe a problem with starting the emulator? Run with `-vs` to see details

**Comment by tsusanka** [comment]
My mistake I've built it without PYOPT=0.

**Comment by tsusanka** [comment]
Looks like this solves https://github.com/trezor/trezor-firmware/issues/243

**Pull request merged**